tests/integration/test_charm.py

```diff
@@ -62,6 +62,55 @@ async def test_prometheus_integration(
         assert len(query_targets["data"]["activeTargets"])
 
 
+@pytest.mark.usefixtures("synapse_app")
+async def test_grafana_integration(
+    model: Model,
+    synapse_app_name: str,
+    prometheus_app_name: str,
+    prometheus_app,  # pylint: disable=unused-argument
+    grafana_app_name: str,
+    grafana_app,  # pylint: disable=unused-argument
+    get_unit_ips: typing.Callable[[str], typing.Awaitable[tuple[str, ...]]],
+):
+    """
+    arrange: after Synapse charm has been deployed.
+    act: establish relations established with grafana charm.
+    assert: grafana Synapse dashboard can be found.
+    """
+    await model.relate(
+        f"{prometheus_app_name}:grafana-source", f"{grafana_app_name}:grafana-source"
+    )
+    await model.relate(synapse_app_name, grafana_app_name)
+
+    await model.wait_for_idle(
+        apps=[synapse_app_name, prometheus_app_name, grafana_app_name],
+        status="active",
+        idle_period=60,
```

Comment:
Why do we need an idle_period of 60 ?

Reply:
Integrating Grafana with Synapse takes some time on "executing", so this idle_period is to be "on the safe side" ensuring that everything is OK before the test.